lib/icingadb/icingadb-utility.cpp

```diff
@@ -245,6 +245,21 @@ String IcingaDB::GetLowerCaseTypeNameDB(const ConfigObject::Ptr& obj)
 }
 
 long long IcingaDB::TimestampToMilliseconds(double timestamp) {
+	// In addition to the limits of the Icinga DB MySQL (0 - 2^64) and PostgreSQL (0 - 2^63) schemata,
+	// years not fitting in YYYY may cause problems, see e.g. https://github.com/golang/go/issues/4556.
+	// RFC 3339: "All dates and times are assumed to be (...) somewhere between 0000AD and 9999AD."
+	//
+	// The below upper limit includes a safety buffer to make sure the timestamp is within 9999AD in all time zones:
+	// $ date -ud @253402214400
+	// Fri Dec 31 00:00:00 UTC 9999
+	// $ TZ=Asia/Vladivostok date -d @253402214400
+	// Fri Dec 31 10:00:00 +10 9999
+	// $ TZ=America/Juneau date -d @253402214400
+	// Thu Dec 30 15:00:00 AKST 9999
+	if (timestamp <= 0.0 || timestamp > 253402214400.0) {
+		return 0;
+	}
```

Comment:
I gave this a try with the following downtime object:

```
object Downtime "yolo" {
	author = "icingaadmin"
	comment = "Test"
	host_name = "master-1"
	start_time = Math.pow(2, 65)
	end_time = Math.pow(2, 66)
}
```

Which then looked like this in Icinga DB Web:

![20240930_14h23m21s_grim](https://github.com/user-attachments/assets/acd102c9-f668-42a7-9a55-ae2cf385df9b)

There's quite a bit of strangeness going on there. "expires in -28794984m" sounds like this would be some broken downtime left-over from the past, when actually, it's absurdly far in the future.

In contrast, I quickly changed to source so that it returns `253402214400 * 1000` on overflow. Yes, it still triggers some rendering glitches and the number of days shown is not accurate, but at least it more accurately shows that it's far in the future:

![20240930_14h20m15s_grim](https://github.com/user-attachments/assets/863e4ae5-be1e-4795-82f5-3fba9baafd8c)

So just like I already suggested in https://github.com/Icinga/icinga2/pull/10058#discussion_r1627587399, clamping the value to the bounds looks like it provides a more useful result.